### DISCUSSION TITLE
How to set the position of the red Rectangle in the navigator in the viewer

### DISCUSSION THREAD

**Discussion by stevendai11** [openseadragon/openseadragon#2124] [Q&A]
Hi, I have a multiple tabbed angular application. One of the tab is for the viewer of the images. However, when I flipped the tab back to the viewer tab from other tabs, the image and the minimap in the navigator show up correctly but the red rectangle is somehow minimized and is set at the top left corner almost like a dot.
Is there a way to reset the position of this red rectangle or FOV (focus of view)?
Thanks a lot for your help.
Steven

**Comment by iangilman** [comment]
Interesting. I imagine what's going on is that the viewer detects that it's 0 size while it's hidden and updates the navigator rectangle to match that and then somehow doesn't realize that it needs to update the navigator rectangle when the viewer pops back to normal size. This seems like a bug. Feel free to file an issue for it if you'd like! 

Meanwhile, if you want to force the navigator to update itself, you should be able to do `viewer.navigator.update()`.

**Reply by stevendai11** [reply]
Thanks for the reply.
viewer.navigator.update() doesn't work. I am using the typescript version so I need to give a viewport as parameter. I tried both navigator and viewer's viewport but no success.
I am not familiar with the bug tracking system so if it's just handy for you, please go ahead and file it.
Thanks again.
Steven

**Reply by iangilman** [reply]
Oh, good point! It should be `viewer.navigator.update(viewer.viewport)`. When are you doing it? You might try adding a small `setTimeout` to it so it happens after everything updates on screen after the viewer is unhidden. 

I've gone ahead and filed an issue: https://github.com/openseadragon/openseadragon/issues/2130. Work here is all volunteer-based, so it'll only get fixed if someone decides to take it on. If you are interested in fixing it, I can help point you in the right direction.

**Comment by stevendai11** [comment]
Thanks a lot for the tip.Setting timeout fixed the problem!Much appreciated.Steven

**Reply by iangilman** [reply]
Excellent :)
